sdk/ai/ai-projects/samples-dev/finetuning/sample_finetuning_reinforcement_job.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * This sample demonstrates how to obtain an OpenAI client and perform reinforcement fine-tuning operations.
+ * Supported OpenAI models: o4-mini
+ *
+ * @summary Using an OpenAI client, this sample demonstrates how to create and cancel reinforcement fine-tuning jobs.
+ */
+
+import { DefaultAzureCredential } from "@azure/identity";
+import { AIProjectClient } from "@azure/ai-projects";
+import type { JobCreateParams } from "openai/resources/fine-tuning/jobs";
+import type { ScoreModelGrader } from "openai/resources/graders/grader-models";
+import { fileURLToPath } from "url";
+import * as fs from "fs";
+import * as path from "path";
+import "dotenv/config";
+
+const projectEndpoint = process.env["AZURE_AI_PROJECT_ENDPOINT"] || "<project endpoint string>";
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const modelName = process.env["MODEL_NAME"] || "o4-mini";
+const trainingFilePath = path.join(__dirname, "data", "rft_training_set.jsonl");
+const validationFilePath = path.join(__dirname, "data", "rft_validation_set.jsonl");
+
+export async function main(): Promise<void> {
+  const project = new AIProjectClient(projectEndpoint, new DefaultAzureCredential());
+
+  const openAIClient = await project.getOpenAIClient();
+  console.log("Created OpenAI client.");
+
+  // 1) Create the training and validation files
+  const trainingFile = await openAIClient.files.create({
+    file: fs.createReadStream(trainingFilePath),
+    purpose: "fine-tune",
+  });
+  console.log(`Uploaded file with ID: ${trainingFile.id}`);
+  const validationFile = await openAIClient.files.create({
+    file: fs.createReadStream(validationFilePath),
+    purpose: "fine-tune",
+  });
+  console.log(`Uploaded file with ID: ${validationFile.id}`);
+
+  // 2) Wait for the files to be processed
+  await openAIClient.files.waitForProcessing(trainingFile.id);
+  await openAIClient.files.waitForProcessing(validationFile.id);
+  console.log("Files processed.");
+
+  const grader: ScoreModelGrader = {
+    name: "Response Quality Grader",
+    type: "score_model",
+    model: "o3-mini",
```

Comment:
The description above mentions o4-mini but the operation uses o3-mini, could this be reconciled?

Reply:
or are we intentionally using two different models, one for training and one for grading?